server/src/main/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResource.java

```diff
@@ -38,170 +62,210 @@
 public class BackfillRowHmacResource {
   private static final Logger logger = LoggerFactory.getLogger(BackfillRowHmacResource.class);
 
-  private final AclDAO aclDAO;
-  private final ClientDAO clientDAO;
-  private final SecretDAO secretDAO;
-  private final SecretContentDAO secretContentDAO;
-  private final SecretSeriesDAO secretSeriesDAO;
+  private final DSLContext jooq;
+  private final RowHmacGenerator rowHmacGenerator;
 
   @Inject
-  public BackfillRowHmacResource(AclDAOFactory aclDAOFactory,
-      ClientDAOFactory clientDAOFactory, SecretDAOFactory secretDAOFactory,
-      SecretContentDAOFactory secretContentDAOFactory,
-      SecretSeriesDAOFactory secretSeriesDAOFactory) {
-    this.aclDAO = aclDAOFactory.readwrite();
-    this.clientDAO = clientDAOFactory.readwrite();
-    this.secretDAO = secretDAOFactory.readwrite();
-    this.secretContentDAO = secretContentDAOFactory.readwrite();
-    this.secretSeriesDAO = secretSeriesDAOFactory.readwrite();
+  public BackfillRowHmacResource(DSLContext jooq, RowHmacGenerator rowHmacGenerator) {
+    this.jooq = jooq;
+    this.rowHmacGenerator = rowHmacGenerator;
   }
 
   /**
    * Backfill row_hmac for this secret.
    */
   @Timed @ExceptionMetered
-  @Path("{secretIdFrom}/{secretIdTo}/backfill-secrets")
+  @Path("backfill-secrets")
   @POST
   @Consumes(APPLICATION_JSON)
   @Produces(APPLICATION_JSON)
-  public boolean backfillSecretsRowHmac(@Auth AutomationClient automationClient,
-      @PathParam("secretIdFrom") Long secretIdFrom, @PathParam("secretIdTo") Long secretIdTo) {
-    int modifiedRows = 0;
-    logger.info("backfill-secrets: processing secrets from {} to {}", secretIdFrom, secretIdTo);
-    for (long secretId = secretIdFrom; secretId < secretIdTo; secretId++) {
-      Optional<SecretSeriesAndContent> secret = secretDAO.getSecretById(secretId);
-
-      if (secret.isEmpty()) {
-        logger.info("backfill-secrets {}: does not exist", secretId);
-        continue;
+  public void backfillSecretsRowHmac() {
```

Comment:
It would be good to be able to skip over a number of records, e.g. if the job crashes at some point you can skip over the records that were already processed.

Reply:
Also, it might be a good idea to have a max ID, so you can ensure that the job ends after processing a number of records, say 1M.